app/Http/Controllers/API/TeamsController.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+namespace App\Http\Controllers\API;
+
+use App\Actions\Teams\CreateTeamAction;
+use App\Actions\Teams\JoinTeamAction;
+use App\Actions\Teams\LeaveTeamAction;
+use App\Actions\Teams\UpdateTeamAction;
+use App\Http\Controllers\Controller;
+use App\Http\Requests\Teams\CreateTeamRequest;
+use App\Http\Requests\Teams\JoinTeamRequest;
+use App\Http\Requests\Teams\LeaveTeamRequest;
+use App\Http\Requests\Teams\UpdateTeamRequest;
+use App\Models\Teams\Team;
+use App\Models\Teams\TeamType;
+use App\Models\User\User;
+use Illuminate\Database\Eloquent\Collection;
+use Illuminate\Support\Facades\Auth;
+
+class TeamsController extends Controller
+{
+
+    public function __construct()
+    {
+        $this->middleware('auth')->except('types');
+    }
+
+    /**
+     * Array of teams the user has joined
+     *
+     * @return array<Team>
+     */
+    public function list()
+    {
+        /** @var User $user */
+        $user = Auth::guard('api')->user();
+
+        return $user->teams;
+    }
+
+    /**
+     * The user wants to create a new team
+     *
+     * @param CreateTeamRequest $request
+     * @param CreateTeamAction $action
+     * @return array
+     */
+    public function create(CreateTeamRequest $request, CreateTeamAction $action): array
+    {
+        /** @var User $user */
+        $user = Auth::guard('api')->user();
+
+        if ($user->remaining_teams === 0) {
+            abort(403, 'You have created your maximum number of teams!');
+        }
+
+        $team = $action->run($user, $request->all());
+
+        return ['team' => $team];
```

Comment:
I think we should return `'success' => true` with every successful response, it makes our app predictable and easy to check for a success response on the backend